### DISCUSSION TITLE
How to handle multiple values per input

### DISCUSSION THREAD

**Discussion by hornta** [react-hook-form/react-hook-form#9135] [Q&A]
We are working on a month range picker that allows the user to pick a start month, 2022-10 and an end month 2023-02. Does it make sense to store both of these values in a single object or do RHF want these as multiple values?

```ts
type MyInputs {
  myDateStart: { year: number, month: number },
  myDateEnd: { year: number, month: number }
}

// vs. 
type MyInputs {
  myDateRange: { 
    start: { year: number, month: number }, 
    end: { year: number, month: number }
  }
}
```

I'm leaning towards the second example but I don't know if RHF works with a value shape like that.

**Comment by bluebill1049** [comment]
I think you want to build a controlled component to host those states and just wrapped with Controller to collect data and validate (if any).
